recipes-core/images/iot2050-image-swu-example.bb

```diff
@@ -28,7 +28,6 @@ WICVARS += "WDOG_TIMEOUT KERNEL_IMAGE INITRD_IMAGE DTB_FILES"
 # not compatible with SWUpdate images
 IMAGE_INSTALL_remove = "regen-rootfs-uuid"
 IMAGE_INSTALL_remove = "install-on-emmc"
-IMAGE_INSTALL_remove = "node-red-preinstalled-nodes"
```

Comment:
That's not that easy, at least as long as you do not disable upgrades from the UI via the palette. Those will strangely fail because they now work against a read-only rootfs. But that would still limit the usability of Node-RED as you can no longer add custom nodes while creating flows.

The idea we had with the Node-RED preinstallation is that users can update also the pre-installed nodes during runtime - that's what they are used to. However, this is naturally in conflict with image-based swupdate. We need to resolve that conflict in a smarter way or leave the issue alone.

Reply:
Ok, got it. However, this only means we still do not include it in the example image. We still can change the install paths so that node-red can be added to a custom swupdate image. We have that use-case and updatability support of the remaining platform is more important for us than Web-UI based updates of node-red modules.

At least as the install paths are now globally changeable in a consistent way, this leaves the decision up to the integrator.

Reply:
I'm not against the other changes, and we could also discuss making the Node-RED palette read-only. Maybe there is a way to tell Node-RED that it cannot change nodes, thus provide useful feedback to users.

In general, users should model custom swupdate images in a way that pre-installs all nodes, not just those that we provide. OTOH, we did not test if our ad-hoc node packaging mechanism really works for most Node-RED nodes. I would not be surprised if at least some fail.

Reply:
Before reworking this please take also a look at this issue:
https://github.com/mindsphere/node-red-contrib-mindconnect/issues/173 

The pre-installed node-red is run for some reason from '/' directory and because the default setting of the mindconnect node is to use `process.cwd() +'/.mc/'` directory for the persistence, which  is now not writeable.

Also, I am not that sure that making node-red and palette read only is such a good idea - every single bugfix (e.g. in custom nodes) would require a new image instead of simple update via palette. This is very inconvenient and the users will probably bug the mainteiners of the nodes instead of you ;)

@fmoessbauer @jan-kiszka we should probably try to get on the call this week to discuss this ;)

Reply:
Happy to explain the background, specifically which setups this is affecting and which not. The more I think about it, the less I believe we should promote manual clicking on the Node-RED palette in case the system is OTA-updated - this is a classic flaw to mix both.

Reply:
Oh, and setting the WorkingDirectory in https://github.com/siemens/meta-iot2050/pull/395/commits/445d77af08aadb3af3a7174fd2122ce9988200f2 should also resolve the current startup issue of mindconnect in a read-only setup (or when we finally de-privilege it).

Reply:
@sn0wcat thanks for cross-referencing this.

> The pre-installed node-red is run for some reason from '/' directory and because the default setting of the mindconnect node is to use process.cwd() +'/.mc/' directory for the persistence, which is now not writeable at all, a fatal error occurs during configuration of the mindsphere connectivity.

Normally applications should store their state either in `/var/` or for user-specific applications below `/home/<user>` (or more precisely `$HOME`). Using the current working directory is highly discouraged and this should be fixed in MCL. However, I don't know other node-modules do it. I guess there is a way to query the [userDir](https://nodered.org/docs/user-guide/runtime/configuration) and all state should be stored there.

> Also, I am not that sure that making node-red and palette read only is such a good idea - every single bugfix (e.g. in custom nodes) would require a new image instead of simple update via palette. 

The whole node-red story is a nightmare from a security perspective anyways. In the swupdate case, we need to ensure to have as little state as possible as otherwise configuration drift happens and updates get unstable. If you want to go that way, the default palette needs to be read-only. However, it still is possible to add more nodes from the palette, but these then have to be updated by the user. This again adds configuration drift with all its side effects.

Reply:
With the latest version, this now fully works (and updates via the palette are still possible).